api/src/api/users/user_routes.py

```diff
@@ -146,3 +148,39 @@ def user_get(db_session: db.Session, user_id: UUID) -> response.ApiResponse:
         return response.ApiResponse(message="Success", data=user)
 
     raise_flask_error(401, "Unauthorized user")
+
+
+@user_blueprint.post("/<uuid:user_id>/saved-opportunities")
+@user_blueprint.input(UserSaveOpportunityRequestSchema, location="json")
+@user_blueprint.output(UserSaveOpportunityResponseSchema)
+@user_blueprint.doc(responses=[200, 401])
+@user_blueprint.auth_required(api_jwt_auth)
+@flask_db.with_db_session()
+def user_save_opportunity(
+    db_session: db.Session, user_id: UUID, json_data: dict
+) -> response.ApiResponse:
+    logger.info("POST /v1/users/:user_id/saved-opportunities")
+
+    user_token_session: UserTokenSession = api_jwt_auth.current_user  # type: ignore
+
+    # Verify the authenticated user matches the requested user_id
+    if user_token_session.user_id != user_id:
+        raise_flask_error(401, "Unauthorized user")
```

Comment:
Completely unrelated to your PR, but if you're looking for something to work on while folks are out, I'd like if we had an AuthZ library or approach that was a bit more sophisticated than this once we have roles: https://github.com/HHS/simpler-grants-gov/issues/2818

Mass PFML had a setup but I looked into the library used and it was not maintained anymore.

Reply:
Good idea, will take a look!